docs/visual-testing/integrations/java.md

```diff
@@ -0,0 +1,172 @@
+---
+sidebar_label: Java
+---
+
+import useBaseUrl from '@docusaurus/useBaseUrl';
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Java WebDriver Integration
+
+## Introduction
+
+This guide requires an existing Java JUnit / TestNG setup.<br />
+You can alternatively take a look to our [example repository](#examples).
+
+Sauce Labs Visual provides an library allowing integration with [WebDriver](https://www.selenium.dev/documentation/webdriver/).
+
+Sauce Labs Visual plugin provides a library exposing a `VisualApi` object that provides actions:
+- `visual.check()`: Takes a screenshot and send it to Sauce Labs Visual for comparison.
+- `visual.checkResults()`: Waits for diff calculations to complete and returns a summary of results.
+
+Sample usage:
+```java
+    assertEquals(visual.checkResults().get(DiffStatus.UNAPPROVED), 2);
+```
+See [definitions](../../visual-testing.md#definitions) to learn more about different diff statuses.
+
+## Quickstart
+
+### Step 1: Add Sauce Labs Visual dependecy
```

Comment:
@kb-kerem 

We have TestNG and JUnit, will the implementation be the same? If not, can we then split this up into two examples?

Reply:
The way we write the test is the same, but the annotations will be different. I'll split it

Reply:
I've split JUnit and TestNG annotation sections (for Before and After) as it's the only difference regarding implementation